packages/react-instantsearch-hooks-server/src/getServerState.tsx

```diff
@@ -57,6 +72,29 @@ export function getServerState(
       return {
         initialResults,
       };
+    })
+    .then((serverState) => {
```

Comment:
```suggestion
    .then((serverState) => {
      // We handle <DynamicWidgets> differently than other components because it
      // requires another query to retrieve the dynamic widgets to render.
```